src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax;
+
+abstract class Figure implements FigureDraw {
+    private Color color;
+
+    public Figure(Color color) {
+        this.color = color;
+    }
+
+    public void setColor(Color color) {

```

Comment:
The setColor() method allows to change the color of the figure after it's creation. It seems unnecessary as color is typically a property that's set upon creation and doesn't change. Consider removing this method, unless there's a specific requirement for it in your program.

Reply:
But then it will be randomised just at the beginning, which means that random function will draw from a set of non randomised objects. Isn't it?